### ISSUE TITLE
[Performance] W8A8 works not so well

### ISSUE THREAD

**Issue by yichuan520030910320** [vllm-project/llm-compressor#1326]
I try to use llm-compressor to quantize a model using W8A8, but the performance even drops

Quant Method
```
from llmcompressor.modifiers.smoothquant import SmoothQuantModifier
from llmcompressor.modifiers.quantization import GPTQModifier
from llmcompressor import oneshot

# Select quantization algorithm. In this case, we:
#   * apply SmoothQuant to make the activations easier to quantize
#   * quantize the weights to int8 with GPTQ (static per channel)
#   * quantize the activations to int8 (dynamic per token)
recipe = [
    SmoothQuantModifier(smoothing_strength=0.8),
    GPTQModifier(scheme="W8A8", targets="Linear", ignore=["lm_head"]),
]

# Apply quantization using the built in open_platypus dataset.
#   * See examples for demos showing how to pass a custom calibration set
oneshot(
    model="facebook/contriever",
    dataset="open_platypus",
    recipe=recipe,
    output_dir="contriever-INT4",
    max_seq_length=2048,
    num_calibration_samples=512,
)

```

Test Method

```
# transformer load /home/ubuntu/Power-RAG/micro/contriever-INT8
from transformers import AutoModelForCausalLM, AutoTokenizer
import torch
import time
import numpy as np

# Set parameters
batch_size = 60
seq_len = 256
num_runs = 10  # Number of runs to average over

# Load INT8 model
int8_model = AutoModelForCausalLM.from_pretrained("/home/ubuntu/Power-RAG/micro/contriever-INT8").cuda()
int8_model.eval()
# Load original FP16 model
torch.set_default_dtype(torch.float16)
fp16_model = AutoModelForCausalLM.from_pretrained(
    "facebook/contriever",
    # device_map="auto"
).cuda()
fp16_model.eval()

# Load tokenizer
tokenizer = AutoTokenizer.from_pretrained("facebook/contriever")

# Create dummy input data (random token IDs)
dummy_input = torch.randint(0, 1000, (batch_size, seq_len))

# Move to appropriate device
device = 'cuda'
dummy_input = dummy_input.to(device)

# Warmup
print("Warming up models...")
with torch.no_grad():
    _ = fp16_model(dummy_input)
    # Move INT8 model to same device if needed
    if hasattr(int8_model, "to"):
        int8_model = int8_model.to(device)
    _ = int8_model(dummy_input)

# Test FP16 model
print(f"Testing FP16 model with batch_size={batch_size}, seq_len={seq_len}")
fp16_times = []
with torch.no_grad():
    for i in range(num_runs):
        start_time = time.time()
        _ = fp16_model(dummy_input)
        torch.cuda.synchronize()  # Make sure GPU operations are completed
        end_time = time.time()
        fp16_times.append(end_time - start_time)
        print(f"Run {i+1}/{num_runs}: {fp16_times[-1]:.4f} seconds")

# Test INT8 model
print(f"Testing INT8 model with batch_size={batch_size}, seq_len={seq_len}")
int8_times = []
with torch.no_grad():
    for i in range(num_runs):
        start_time = time.time()
        _ = int8_model(dummy_input)
        torch.cuda.synchronize()  # Make sure GPU operations are completed
        end_time = time.time()
        int8_times.append(end_time - start_time)
        print(f"Run {i+1}/{num_runs}: {int8_times[-1]:.4f} seconds")

# Calculate statistics
fp16_mean = np.mean(fp16_times)
fp16_std = np.std(fp16_times)
int8_mean = np.mean(int8_times)
int8_std = np.std(int8_times)
speedup = fp16_mean / int8_mean

# Print results
print("\n===== RESULTS =====")
print(f"FP16 Model: {fp16_mean:.4f} ± {fp16_std:.4f} seconds")
print(f"INT8 Model: {int8_mean:.4f} ± {int8_std:.4f} seconds")
print(f"Speedup: {speedup:.2f}x")
print("==================")
```

Result

```
===== RESULTS =====
FP16 Model: 0.0732 ± 0.0037 seconds
INT8 Model: 0.1883 ± 0.0003 seconds
Speedup: 0.39x
==================
```

**Comment by dsikka** [comment]
Hi @yichuan520030910320 

It seems like you’re comparing speed here.
Once quantized, to leverage the model in its compressed form, you’ll need to use vLLM. 

llmcompressor does not support compressed generation. It runs forward passes in the model’s decompressed form in order and can be used to evaluate accuracy. However, as the model is not running compressed, no speedup is expected. 

Please let us know if you need help running vLLM.
